Added Debian and Ubuntu

austgate opened the pull request:
I have added Debian and Ubuntu to the parameters list and spec tests.

austgate committed:
Updated params.pp to include Debian/Ubuntu arguments to install sysstat

austgate committed:
Added Debian & Ubuntu tests to sysstat_params_spec.rb

jhoblitt commented on manifests/params.pp:
```diff
@@ -12,7 +12,7 @@
 class sysstat::params {
   case $::osfamily {
-    'redhat': {
+    'debian', 'ubuntu, 'redhat': {
```
Travis is failing as there's a missing single quote after 'ubuntu.

jhoblitt replied:
Also note that Ubuntu is not an osfamily.  $::operatingsystem == Ubuntu is part of $::osfamily == Debian.

austgate replied:
@jhoblitt I'll make that change to the pull request.

jhoblitt commented on spec/classes/sysstat_params_spec.rb:
```diff
@@ -11,11 +11,31 @@
-        :osfamily        => 'Debian',
-        :operatingsystem => 'Debian',
+        :osfamily        => 'Fedora',
+        :operatingsystem => 'Fedora',
```
While it would technically work for this test, Fedora is not an osfamily and it is part of osfamily == RedHat, so it would be better to use another family name like Solaris, Suse, etc. to avoid confusion about Fedora being supported (or not).

austgate replied:
Okay, I'll make that change.

jhoblitt commented:
@austgate ping?

austgate committed:
Update params.pp

Ubuntu removed from OSFamily as per comments from @jhoblitt

austgate committed:
Update sysstat_params_spec.rb

Changed the incorrect Fedora to Solaris :osfamily and :operatingsystem. Changed error message to "not supported on Solaris" from "not supported on Debian"

austgate commented:
Sorry for delay in replying, don't appear to have received the earlier notifications. I've update the tests and the osfamily in this pull request as per your comments.

jhoblitt commented:
@austgate This looks great -- thank you for the contribution!  I'm going to merge this in a moment but with your 4 commits squashed into one (you will still be the author of record).  You'll likely want to replace your master branch with mine or you'll run into divergent history problems.

While there's certainly more than one way do to it, there's a few general best practices that have grown up around PRs on github.  Hopefully, you don't already know this and I'm not putting my foot in my mouth :)
- Commit messages normally are formatted as an `imperative` statement.  See: https://github.com/erlang/otp/wiki/writing-good-commit-messages#do
- PRs are typically made from a branch other than your own master.  There's nothing wrong with making a PR from master but it prevents you from modifying that branch until the PR is resolved.
- There's still active debate over the usage of rebase on feature/topic branches but I believe that the major of folks are now editing history (as I will do for your PR), at least for small to medium sized branches.

jhoblitt commented:
@austgate I've pushed a new release to the forge that incorporates your PR: http://forge.puppetlabs.com/jhoblitt/sysstat/1.1.0

Thanks again for contributing!
